### DISCUSSION TITLE
Update dependencies

### DISCUSSION THREAD

**Discussion by fangmarks** [zeit-ui/react#473] [General]
Would it be possible to update Geist to the latest react version?

I want to use it in my project, but it's using react v17, Geist still uses v16.14, which means I'm not able to use Geist

**Comment by unix** [comment]
Geist releases packages that do not depend on React, of course, the `master` branch code is still based on the `react: ^17.0.1` build.
Or did I miss any breaking updates?

**Reply by fangmarks** [reply]
I'm not sure, but then I try to add Geist into the [NextJS with Typescript](https://github.com/vercel/next.js/tree/canary/examples/with-typescript) Example and then run `rpm ls react`, it tells me that `@geist-ui/react` still requires 16.14 through `styled-jsx`

![image](https://user-images.githubusercontent.com/20520193/110657715-5a626080-81c1-11eb-8497-83a8dd1e31d5.png)


If I then try to import Geist components anywhere it will just crash

**Reply by unix** [reply]
I got it. 
There was an [issue styled-jsx#696](https://github.com/vercel/styled-jsx/issues/696) in `styled-jsx@3.4.x` that prevented us from upgrading immediately, which is the reason React is stuck at `v16`.

This [issue styled-jsx#695](https://github.com/vercel/styled-jsx/issues/695) discusses some details of the problem, but I haven't found a good fix so far.

Now that the `styled-jsx` repository is not well maintained and there are other issues that have not been resolved, I need to evaluate the effort of replacing the JSS solution.  (In the short term we cannot upgrade the dependency library)

**Reply by mathewgurney** [reply]
Just to clarify, is Geist not currently compatible with the latest versions of react because of the issues with styled-jsx?
